src/modules/github/components/pulls-review-dialog/pulls-review-dialog.tsx

```diff
@@ -0,0 +1,262 @@
+
+import { EPullRequestType } from '@/types/github/root/index'
+
+import { useSubmitForm } from './hooks'
+import { getFlattenCheckedPulls } from './utils'
+
+interface IPullsReviewDialogProps {
+  isDialogOpen: boolean
+  handleSetIsOpenDialog: (open: boolean) => void
+  handleSetHasChecked: (value: boolean) => void
+}
+
+export const PullsReviewDialog = ({
+  isDialogOpen,
+  handleSetIsOpenDialog,
+  handleSetHasChecked
+}: IPullsReviewDialogProps) => {
+  const [commentInput, setCommentInput] = useState<string>('')
+  const [focusIndex, setFocusIndex] = useState<number>(0)
+  const [radioButtonValue, setRadioButtonValue] = useState<EPullRequestType>(
+    EPullRequestType.COMMENT
+  )
+  const { repos } = useRepos()
+  const { translate } = useDictionary()
+  const {
+    progressData,
+    submit,
+    isLoading,
+    error: errors,
+    reset
+  } = useSubmitForm()
+  const radioButtonValues = [
+    {
+      value: EPullRequestType.COMMENT,
+      label: translate('GITHUB.PULL_REVIEW_FORM_COMMENT_BUTTON')
+    },
+    {
+      value: EPullRequestType.APPROVE,
+      label: translate('GITHUB.PULL_REVIEW_FORM_APPROVE_BUTTON')
+    },
+    {
+      value: EPullRequestType.REQUEST_CHANGES,
+      label: translate('GITHUB.PULL_REVIEW_FORM_REQUEST_CHANGES_BUTTON')
+    }
+  ]
+  const flattenCheckedPulls = getFlattenCheckedPulls(repos)
+  const focusedPull =
+    flattenCheckedPulls.length > 0 ? flattenCheckedPulls[focusIndex] : undefined
+
+  const handleRadioButtonChange = (value: string) => {
+    setRadioButtonValue(value as EPullRequestType)
+  }
+
+  const handleCommentChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    setCommentInput(event.target.value)
+  }
+
+  const handleOpenChange = (open: boolean) => {
+    setFocusIndex(0)
+    handleSetIsOpenDialog(open)
+  }
+
+  const handleCancelClick = () => {
+    handleSetIsOpenDialog(false)
+    setFocusIndex(0)
+  }
+
+  const handleLeftClick = () => {
+    if (focusIndex === 0) {
+      return
+    }
+    setFocusIndex((prev) => prev - 1)
+  }
+
+  const handleRightClick = () => {
+    if (focusIndex === flattenCheckedPulls.length - 1) {
+      return
+    }
+
+    setFocusIndex((prev) => prev + 1)
+  }
+
+  const handleSubmit = () => {
+    submit({
+      checkedPulls: flattenCheckedPulls,
+      reviewType: radioButtonValue,
+      comment: commentInput
+    })
+      .then(() => {
+        handleSetIsOpenDialog(false)
+      })
+      .catch(console.error)
+  }
+
+  useEffect(() => {
+    if (!isDialogOpen) {
+      reset()
+    }
+  }, [isDialogOpen])
+
+  useEffect(() => {
+    if (flattenCheckedPulls.length > 0) {
+      handleSetHasChecked(true)
+    } else {
+      handleSetHasChecked(false)
+    }
+  }, [flattenCheckedPulls.length])
+
+  // This is not gonna happen.
+  if (!focusedPull) {
+    return <></>
+  }
+
+  return (
+    <Dialog
+      open={isDialogOpen}
+      onOpenChange={handleOpenChange}
+      title={focusedPull.pullTitle}
+      widthType={'default'}
```

Comment:
Wrong.
`widthType="default"`

Reply:
lol I already modified, Probably overlapped in the same time zone

Reply:
OMG okay